apis/supernova/src/handler/utils/hypercube-helper/hypercube-utils.js

```diff
@@ -98,24 +88,6 @@ function insertMainDimension(self, dimension, dimensions, idx) {
   });
 }
 
-export function addMainDimension(self, dimension, index) {
-  const dimensions = self.getDimensions();
-  const idx = index ?? dimensions.length;
-
-  if (dimensions.length < self.maxDimensions()) {
-    return insertMainDimension(self, dimension, dimensions, idx);
-  }
-
-  return Promise.resolve();
-}
```

Comment:
Moved it to a separated file `add-main-dimension.js` because of having a nested function